mlflow/tracing/export/mlflow.py

```diff
@@ -94,26 +86,17 @@ def _export_trace(self, root_span: MlflowSpanWrapper):
         # TODO: Make this async
         self._client.log_trace(trace)
 
-    def _serialize_inputs_outputs(self, input_or_output: Optional[Any]) -> str:
+    def _truncate_metadata(self, value: Optional[str]) -> str:
         """
-        Serialize inputs or outputs field of the span to a string, and truncate if necessary.
+        Get truncated value of the attribute if it exceeds the maximum length.
         """
-        if not input_or_output:
-            return ""
-
-        try:
-            serialized = json.dumps(input_or_output, default=str)
-        except TypeError:
-            # If not JSON/string serializable, raise a warning and return an empty string
-            _logger.warning(
-                "Failed to serialize inputs/outputs for a trace, an empty string will be recorded."
-            )
+        if not value:
             return ""
 
-        if len(serialized) > MAX_CHARS_IN_TRACE_INFO_METADATA_AND_TAGS:
+        if len(value) > MAX_CHARS_IN_TRACE_INFO_METADATA_AND_TAGS:
             trunc_length = MAX_CHARS_IN_TRACE_INFO_METADATA_AND_TAGS - len(TRUNCATION_SUFFIX)
-            serialized = serialized[:trunc_length] + TRUNCATION_SUFFIX
-        return serialized
+            value = value[:trunc_length] + TRUNCATION_SUFFIX
```

Comment:
just curious, if an attribute value is a JSON object, might this truncation cause it to become invalid JSON?

Reply:
Great point, probably the question is whether we gonna deserialize it or not? Per my understanding they will be only used to show some digest information of the Trace in UI, so we don't necessarily deserialize it back to Python object.

Reply:
Hmm yeah that makes sense to me. It might make it more difficult to pretty print in the trace UI but we can always just render it as a string if we fail to parse it as JSON

Reply:
Yes, I might wrong as I'm not super familiar with what will be the UI for showing those `TraceInfo` fields tho.

Reply:
Great question! We aren't showing the full value in the trace UI or rendering as JSON, we're just showing the first 250 chars so that users can visually distinguish various rows. cc @hubertzub-db